packages/matter-node.js/test/persistence/JsonConverterTest.ts

```diff
@@ -0,0 +1,106 @@
+
+            assert.equal(json, "{\"aNumber\":1,\"aNumberString\":\"2\",\"aString\":\"hello\",\"aBoolean\":true,\"aNull\":null,\"anObject\":{\"a\":1,\"b\":2,\"c\":3},\"anNumberArray\":[1,2,3],\"anStringArray\":[\"a\",\"b\",\"c\"],\"anObjectArray\":[{\"a\":1},{\"b\":2},{\"c\":3}]}");
+
+            const decodedObj = fromJson(json);
+
+            assert.deepEqual(decodedObj, obj);
+        });
+
+        it("encode/decode BigInt", () => {
+            const obj = BigInt("12345678901234567890");
+
+            const json = toJson(obj);
+
+            assert.equal(json, "\"{\\\"__object__\\\":\\\"BigInt\\\",\\\"__value__\\\":\\\"12345678901234567890\\\"}\"");
+
+            const decodedObj = fromJson(json);
+
+            assert.deepEqual(decodedObj, obj);
+            assert.equal(typeof decodedObj, "bigint");
+        });
+
+        it("encode/decode Buffer", () => {
+            const obj = Buffer.from("hello");
+
+            const json = toJson(obj);
+
+            assert.equal(json, "\"{\\\"__object__\\\":\\\"Buffer\\\",\\\"__value__\\\":\\\"aGVsbG8=\\\"}\"");
+
+            const decodedObj = fromJson(json);
+
+            assert.deepEqual(decodedObj, obj);
+            assert.ok(decodedObj instanceof Buffer);
+        });
+
+        it("encode/decode Uint8Array", () => {
+            const obj = new Uint8Array([1, 2, 3]);
+
+            const json = toJson(obj);
+
+            assert.equal(json, "\"{\\\"__object__\\\":\\\"Uint8Array\\\",\\\"__value__\\\":\\\"AQID\\\"}\"");
+
+            const decodedObj = fromJson(json);
+
+            assert.deepEqual(decodedObj, obj);
+            assert.ok(decodedObj instanceof Uint8Array);
+        });
+
+        it("encode/decode Map", () => {
+            const obj = new Map([["a", 1], ["b", 2], ["c", 3]]);
+
+            const json = toJson(obj);
+
+            assert.equal(json, "\"{\\\"__object__\\\":\\\"Map\\\",\\\"__value__\\\":\\\"[[\\\\\\\"a\\\\\\\",1],[\\\\\\\"b\\\\\\\",2],[\\\\\\\"c\\\\\\\",3]]\\\"}\"");
+
+            const decodedObj = fromJson(json);
+
+            assert.deepEqual(decodedObj, obj);
+            assert.ok(decodedObj instanceof Map);
+            assert.equal(decodedObj.get("a"), 1);
+        });
+
+        it("encode/decide Map of Maps", () => {
+            const obj = new Map([[1, new Map([["a", 1]])], [2, new Map([["b", 2]])]]);
+
+            const json = toJson(obj);
+
+            assert.equal(json, "\"{\\\"__object__\\\":\\\"Map\\\",\\\"__value__\\\":\\\"[[1,\\\\\\\"{\\\\\\\\\\\\\\\"__object__\\\\\\\\\\\\\\\":\\\\\\\\\\\\\\\"Map\\\\\\\\\\\\\\\",\\\\\\\\\\\\\\\"__value__\\\\\\\\\\\\\\\":\\\\\\\\\\\\\\\"[[\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\"a\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\",1]]\\\\\\\\\\\\\\\"}\\\\\\\"],[2,\\\\\\\"{\\\\\\\\\\\\\\\"__object__\\\\\\\\\\\\\\\":\\\\\\\\\\\\\\\"Map\\\\\\\\\\\\\\\",\\\\\\\\\\\\\\\"__value__\\\\\\\\\\\\\\\":\\\\\\\\\\\\\\\"[[\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\"b\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\",2]]\\\\\\\\\\\\\\\"}\\\\\\\"]]\\\"}\"");
```

Comment:
Could you use  \` to quote the string w/quotes and avoid backslashes?

Reply:
Adjusted and simplified one level ...

Reply:
It definitely looks better, but is there a way to get rid of the need for backslashes altogether if you use backtick `\`` or single-quote `'` around the outer string?

Reply:
No, because they are there because it is a Json inside a JSON ... The idea of the encoding of these special values was to encode them as their own JSON ... so you end up in a "double encoding" when you stringify that again ... but so we are sure that the "special encoded values" are not touched by anything else ... 

We do that in the office if we have values that should be just passed through without anyone else should touch the - simpyl pass as string and done. But that approach in this case does "string in string" and so these backslashes are real :-)